### PR TITLE
Enable queuing_budget in Workflow Metadata

### PR DESCRIPTION
**Pull request by migueltol22**
# TL;DR
Allow a user to specify `queueing_budget` parameter for their workflow. This parameter corresponds to the amount of time a user is willing to wait for a workflow to be executed.

FlyteIDL PR: https://github.com/lyft/flyteidl/pull/50
## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/245

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/models/core/workflow.py [hunk 1]
```diff
@@ -446,25 +446,35 @@ def from_flyte_idl(cls, pb2_object):
 
 class WorkflowMetadata(_common.FlyteIdlEntity):
 
-    def __init__(self):
+    def __init__(self, queuing_budget=None):
         """
         Metadata for the workflow.
```
Comment (wild-endeavor) [inline flytekit/models/core/workflow.py:451]: add a typehint in the docstring?

File: flytekit/common/workflow.py [hunk 2]
```diff
@@ -441,17 +441,18 @@ def _discover_workflow_components(workflow_class):
     return inputs, outputs, nodes
 
 
-def build_sdk_workflow_from_metaclass(metaclass, cls=None):
+def build_sdk_workflow_from_metaclass(metaclass, queuing_budget=None, cls=None):
     """
     :param T metaclass:
     :param cls: This is the class that will be instantiated from the inputs, outputs, and nodes.  This will be used
```
Comment (wild-endeavor) [inline flytekit/common/workflow.py:447]: same here, add a comment for the added param.

File: flytekit/models/core/workflow.py [hunk 3]
```diff
@@ -446,25 +446,35 @@ def from_flyte_idl(cls, pb2_object):
...
+        return self._queuing_budget
 
     def to_flyte_idl(self):
         """
         :rtype: flyteidl.core.workflow_pb2.WorkflowMetadata
         """
-        return _core_workflow.WorkflowMetadata()
+        workflow_metadata = _core_workflow.WorkflowMetadata()
+        if self._queuing_budget:
+            workflow_metadata.queuing_budget.FromTimedelta(self.queuing_budget)
```
Comment (wild-endeavor) [inline flytekit/models/core/workflow.py:468]: this sets the value?

File: flytekit/sdk/workflow.py [hunk 4]
```diff
@@ -42,7 +42,7 @@ def __init__(self, value, sdk_type=None, help=None):
         )
 
 
-def workflow_class(_workflow_metaclass=None, cls=None):
+def workflow_class(_workflow_metaclass=None, cls=None, queuing_budget=None):
     """
```
Comment (wild-endeavor) [inline flytekit/sdk/workflow.py:46]: add param comment

File: flytekit/models/core/workflow.py (hunk 3 above)
Comment (migueltol22) [reply flytekit/models/core/workflow.py:468]: That is correct.

File: flytekit/models/core/workflow.py (hunk 1 above)
Comment (migueltol22) [reply flytekit/models/core/workflow.py:451]: done.

File: flytekit/__init__.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
 from __future__ import absolute_import
 import flytekit.plugins
 
-__version__ = '0.7.0b1'
+__version__ = '0.7.0b2'
```
Comment (wild-endeavor) [inline flytekit/__init__.py:4]: b3 please.

File: tests/flytekit/unit/models/core/test_workflow.py [hunk 6]
```diff
@@ -29,7 +29,7 @@ def test_alias():
 
 
 def test_workflow_metadata():
-    obj = _workflow.WorkflowMetadata()
+    obj = _workflow.WorkflowMetadata(queuing_budget=timedelta(seconds=10))
```
Comment (wild-endeavor) [inline tests/flytekit/unit/models/core/test_workflow.py:32]: can you add another test that does the whole .from_flyte_idl(blah.to_flyte_idl()) tests?  there should be some in this file.  Can you test both the case where the metadata is None and when it's not?

Comment (migueltol22) [reply tests/flytekit/unit/models/core/test_workflow.py:93]: Done.
